demo

```diff
@@ -133,28 +137,59 @@ docker-compose \
     -f docker-compose.yml \
     -f docker-compose.storage.minio.yml \
     -f docker-compose.demo.yml \
+    -p ${DOCKER_COMPOSE_PROJECT_NAME} \
     $CLIENT \
     "$@" -d
 
+retval=$?
+if [[ $retval -ne 0 ]]; then
+    echo "Failed to start docker compose"
+    exit $retval
+fi
+
 USER='mender-demo@example.com'
-PASSWORD=$(< /dev/urandom 2>/dev/null tr -dc A-Za-z0-9 | head -c12)
+PASSWORD=$(base64 /dev/urandom | head -c 12)
 
-MENDER_SERVER_USER="${USER}:${PASSWORD}"
-MENDER_SERVER_URI="https://localhost"
 
 RETRY_LIMIT=5
 
+# Get the IP of the mender-api-gateway
+RETRIES=0
+while :
+do
+    MENDER_TEST_GATEWAY=$(docker ps -q --filter name=${DOCKER_COMPOSE_PROJECT_NAME}_mender-api-gateway | xargs docker inspect --format={{.NetworkSettings.Networks.${DOCKER_COMPOSE_PROJECT_NAME}_mender.IPAddress}})
+    if [[ $MENDER_TEST_GATEWAY == "" ]]; then
+        echo "The mender-gateway-api service does not seem to be up yet. Retrying..."
+    else
+        echo "Mender gateway IP address: " $MENDER_TEST_GATEWAY
+        break
+    fi
+    if [[ RETRIES -eq 5 ]]; then
+        echo "Failed to get the IP of the 'mender-api-gateway' service"
+        exit 1
+    fi
+    RETRIES=$((RETRIES+1))
+    sleep 5
+done
+
+MENDER_SERVER_URI="https://${MENDER_TEST_GATEWAY:-localhost}"
+
+# On MacOS docker is run in a VM,
+# therefore curl calls to the container
+# IP address does not reach the containers
+# like they should. Therefore, default to localhost.
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    MENDER_SERVER_URI="https://localhost"
+fi
```

Comment:
Do we actually need the non-Darwin variant here? Can't we just always connect to localhost?

Reply:
Will that not send the request to all the running compose setups when the integration tests are run?

Reply:
It shouldn't. The line [is removed](https://github.com/mendersoftware/integration/blob/master/tests/run.sh#L53) when running the tests, so only the actual demo test will use it.

Reply:
Ahh, beautiful! Paranoia started gripping me halfway through last week, so I started removing anything that could make it interfere with the other tests